### PR TITLE
tests/service/iot: Implement remaining sweepers

### LINKED ISSUE #14963: Add sweeper for IoT certificates
### Description

Implement a sweeper for IoT certificates..

### Affected Resource

* aws_iot_certificate

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #14963

The `aws_iot_certificate` resource sweeper required some additional resources, so just did all the remaining IoT resources.

Output from sweeper in AWS Commercial:

```
2021/04/07 15:02:21 [DEBUG] Running Sweepers for region (us-west-2):
2021/04/07 15:02:21 [DEBUG] Running Sweeper (aws_iot_policy_attachment) in region (us-west-2)
2021/04/07 15:02:25 [DEBUG] Running Sweeper (aws_iot_thing_principal_attachment) in region (us-west-2)
2021/04/07 15:02:29 [DEBUG] Running Sweeper (aws_iot_certificate) in region (us-west-2)
2021/04/07 15:02:29 [DEBUG] Running Sweeper (aws_iot_thing) in region (us-west-2)
2021/04/07 15:02:30 [DEBUG] Deleting IoT Thing: {
  ThingName: "tf-acc2-267621613065946261"
}
2021/04/07 15:02:30 [DEBUG] Deleting IoT Thing: {
  ThingName: "tf-acc2-3832025121855435663"
}
2021/04/07 15:02:30 [DEBUG] Deleting IoT Thing: {
  ThingName: "tf-acc-2734006379715404679"
}
2021/04/07 15:02:30 [DEBUG] Deleting IoT Thing: {
  ThingName: "tf-acc-7731040170511226170"
}
2021/04/07 15:02:30 [DEBUG] Deleting IoT Thing: {
  ThingName: "tf-acc2-4816803372918593609"
}
2021/04/07 15:02:30 [DEBUG] Running Sweeper (aws_iot_thing_type) in region (us-west-2)
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_6847254873577260292"
}
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_type_96nqxu17"
}
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_type_8o6esytm"
}
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_8755625355827090279"
}
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_5511629522525982263"
}
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_type_xs7cue1y"
}
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_2923393114004902330"
}
2021/04/07 15:02:31 [DEBUG] Deprecating IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_6106631190203262926"
}
2021/04/07 15:02:31 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_type_96nqxu17"
}
2021/04/07 15:02:31 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_5511629522525982263"
}
2021/04/07 15:02:31 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_6847254873577260292"
}
2021/04/07 15:02:31 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_8755625355827090279"
}
2021/04/07 15:02:31 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_2923393114004902330"
}
2021/04/07 15:02:31 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_type_xs7cue1y"
}
2021/04/07 15:02:32 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_type_8o6esytm"
}
2021/04/07 15:02:32 [DEBUG] Deleting IoT Thing Type: {
  ThingTypeName: "tf_acc_iot_thing_type_6106631190203262926"
}
2021/04/07 15:07:43 [DEBUG] Running Sweeper (aws_iot_policy) in region (us-west-2)
2021/04/07 15:07:45 [DEBUG] Running Sweeper (aws_iot_role_alias) in region (us-west-2)
2021/04/07 15:07:45 Sweeper Tests ran successfully:
        - aws_iot_thing_principal_attachment
        - aws_iot_certificate
        - aws_iot_thing
        - aws_iot_thing_type
        - aws_iot_policy
        - aws_iot_role_alias
        - aws_iot_policy_attachment
2021/04/07 15:07:45 [DEBUG] Running Sweepers for region (us-east-1):
2021/04/07 15:07:45 [DEBUG] Running Sweeper (aws_iot_policy_attachment) in region (us-east-1)
2021/04/07 15:07:46 [DEBUG] Running Sweeper (aws_iot_thing_principal_attachment) in region (us-east-1)
2021/04/07 15:07:46 [DEBUG] Running Sweeper (aws_iot_certificate) in region (us-east-1)
2021/04/07 15:07:46 [DEBUG] Running Sweeper (aws_iot_thing) in region (us-east-1)
2021/04/07 15:07:46 [DEBUG] Running Sweeper (aws_iot_thing_type) in region (us-east-1)
2021/04/07 15:07:46 [DEBUG] Running Sweeper (aws_iot_policy) in region (us-east-1)
2021/04/07 15:07:46 [DEBUG] Running Sweeper (aws_iot_role_alias) in region (us-east-1)
2021/04/07 15:07:46 Sweeper Tests ran successfully:
        - aws_iot_thing
        - aws_iot_thing_type
        - aws_iot_policy
        - aws_iot_role_alias
        - aws_iot_policy_attachment
        - aws_iot_thing_principal_attachment
        - aws_iot_certificate
ok      github.com/terraform-providers/terraform-provider-aws/aws       331.909s
```

Output from sweeper in AWS GovCloud (US):

```
2021/04/07 15:02:27 [DEBUG] Running Sweepers for region (us-gov-west-1):
2021/04/07 15:02:27 [DEBUG] Running Sweeper (aws_iot_thing_principal_attachment) in region (us-gov-west-1)
2021/04/07 15:02:30 [DEBUG] Running Sweeper (aws_iot_thing) in region (us-gov-west-1)
2021/04/07 15:02:31 [DEBUG] Running Sweeper (aws_iot_policy_attachment) in region (us-gov-west-1)
2021/04/07 15:02:31 [DEBUG] Running Sweeper (aws_iot_policy) in region (us-gov-west-1)
2021/04/07 15:02:32 [DEBUG] Running Sweeper (aws_iot_certificate) in region (us-gov-west-1)
2021/04/07 15:02:32 [DEBUG] Running Sweeper (aws_iot_role_alias) in region (us-gov-west-1)
2021/04/07 15:02:32 [DEBUG] Running Sweeper (aws_iot_thing_type) in region (us-gov-west-1)
2021/04/07 15:02:33 Sweeper Tests ran successfully:
        - aws_iot_role_alias
        - aws_iot_thing_type
        - aws_iot_thing_principal_attachment
        - aws_iot_thing
        - aws_iot_policy_attachment
        - aws_iot_policy
        - aws_iot_certificate
ok      github.com/terraform-providers/terraform-provider-aws/aws       8.594s
```

### CODE REVIEW

**Review by YakDriver — APPROVED** [review]
Looking good! 🎉

File: aws/resource_aws_iot_certificate_test.go [hunk 1]
```diff
@@ -2,15 +2,76 @@ package aws
...
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing IoT Certificate for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Certificate for %s: %w", region, err))
+		}
+	}
```
Comment (bflad) [inline aws/resource_aws_iot_certificate_test.go:65]:
```suggestion
	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
		errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Certificate for %s: %w", region, err))
	}
```

File: aws/resource_aws_iot_policy_attachment_test.go [hunk 2]
```diff
@@ -2,15 +2,92 @@ package aws
...
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing IoT Policy Attachment for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Policy Attachment for %s: %w", region, err))
+		}
+	}
```
Comment (bflad) [inline aws/resource_aws_iot_policy_attachment_test.go:81]:
```suggestion
	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
		errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Policy Attachment for %s: %w", region, err))
	}
```

File: aws/resource_aws_iot_policy_test.go [hunk 3]
```diff
@@ -2,16 +2,76 @@ package aws
...
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing IoT Policy for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Policy for %s: %w", region, err))
+		}
+	}
```
Comment (bflad) [inline aws/resource_aws_iot_policy_test.go:65]:
```suggestion
	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
		errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Policy for %s: %w", region, err))
	}
```

File: aws/resource_aws_iot_role_alias_test.go [hunk 4]
```diff
@@ -2,15 +2,73 @@ package aws
...
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing IoT Role Alias for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Role Alias for %s: %w", region, err))
+		}
+	}
```
Comment (bflad) [inline aws/resource_aws_iot_role_alias_test.go:62]:
```suggestion
	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
		errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Role Alias for %s: %w", region, err))
	}
```

File: aws/resource_aws_iot_thing_principal_attachment_test.go [hunk 5]
```diff
@@ -2,15 +2,92 @@ package aws
...
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing IoT Thing Principal Attachment for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Thing Principal Attachment for %s: %w", region, err))
+		}
+	}
```
Comment (bflad) [inline aws/resource_aws_iot_thing_principal_attachment_test.go:81]:
```suggestion
	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
		errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Thing Principal Attachment for %s: %w", region, err))
	}
```

File: aws/resource_aws_iot_thing_test.go [hunk 6]
```diff
@@ -2,15 +2,73 @@ package aws
...
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing IoT Thing for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Thing for %s: %w", region, err))
+		}
+	}
```
Comment (bflad) [inline aws/resource_aws_iot_thing_test.go:62]:
```suggestion
	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
		errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Thing for %s: %w", region, err))
	}
```

File: aws/resource_aws_iot_thing_type_test.go [hunk 7]
```diff
@@ -2,15 +2,73 @@ package aws
...
+
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("error listing IoT Thing Type for %s: %w", region, err))
+	}
+
+	if len(sweepResources) > 0 {
+		if err := testSweepResourceOrchestrator(sweepResources); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Thing Type for %s: %w", region, err))
+		}
+	}
```
Comment (bflad) [inline aws/resource_aws_iot_thing_type_test.go:62]:
```suggestion
	if err := testSweepResourceOrchestrator(sweepResources); err != nil {
		errs = multierror.Append(errs, fmt.Errorf("error sweeping IoT Thing Type for %s: %w", region, err))
	}
```

**Review by bflad — COMMENTED** [review]
Removing `len()` check like the other PRs 👍

**Pull request merged**